chainer/functions/array/transpose.py

```diff
@@ -0,0 +1,44 @@
+import numpy
+
+from chainer import function
+from chainer.utils import type_check
+
+
+class Transpose(function.Function):
+    """Permute the dimensions of an array."""
+
+    def __init__(self, axes):
+        self.axes = axes
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 1,)
+
+    @property
+    def label(self):
+        return 'Transpose'
+
+    def forward(self, inputs):
+        x = inputs[0]
+        y = x.transpose(self.axes)
+        return y,
+
+    def backward(self, inputs, grad_outputs):
+        gy = grad_outputs[0]
+        invAxes = numpy.argsort(self.axes)
+        gx = gy.transpose(invAxes)
```

Comment:
Currently, it seems CuPy's transpose does not support numpy.ndarray argument. Use a tuple instead.